### PR TITLE
Update free-https.md

### LINKED ISSUE #2460: Can sites using Free HTTPS Service also use CloudFlare?
Re: https://pantheon.io/docs/free-https/
## Issue Description
With the addition of Free HTTPS some users have questions about how to configure the site with Cloudflare

## Suggested Resolution
add in CloudFlare config information

### PR DESCRIPTION
**Pull request by lauren-kelly**
Closes #2460 

## Effect
PR includes the following changes:
- updates users on Cloudflare Configuration options

## Remaining Work
- [x] verify that using CloudFlare CDN will require users to use CNAME records for both bare and subdomains, rather than following the dashboard recommendations

### CODE REVIEW

**Comment by lauren-kelly** [comment]
@rachelwhitton actually shouldn't have put in the PR yet - let's wait for @ari-gold to confirm the DNS config

File: source/_docs/free-https.md [hunk 1]
```diff
@@ -139,6 +139,14 @@ No, after you update your DNS records, traffic will gracefully switch over and i
...
 All modern browsers and operating systems are supported. For details, see the **Handshake Simulation** portion of this [report](https://www.ssllabs.com/ssltest/analyze.html?d=pantheon.io).
 
+### Can I continue to use CloudFlare 
+Yes, you can continue to use Cloudflare. 
+Unless you require a special feature from Cloudflare (e.g. WAF) then we recommend using Cloudflare for DNS-only:
+  - add your domains, matching the recommendations on your Dashboard
+  - click the cloud to the right of the domain, to turn it grey, rather than orange. This will cause the domain traffic to go straight to Pantheon, bypassing CloudFlare's services
+
+You can also keep using Cloudflare as a CDN on top of the Pantheon CDN, following our [CloudFlare configuration recommendations](/docs/guides/cloudflare-enable-https/#dns-configuration) in which case the Pantheon Domains tool will show action required because the DNS detected will be Cloudflare IP addresses and not match our recommendations. As long as your DNS is no longer routing to a legacy loadbalancer, you can safely ignore the action required warning.
+
```
Comment (rachelwhitton) [inline source/_docs/free-https.md]: I haven't found a way to configure DNS to keep using Cloudflare as a CDN on top of the Pantheon CDN. CNAME flattening results in `ERR_SSL_PROTOCOL_ERROR` 

Using Cloudflare for just DNS using the recommendations from the Pantheon Site Dashboard works as expected and the Domains tool confirms successful config.

**Comment by lauren-kelly** [comment]
@rachelwhitton were you able to get this working with leaving the CNAME configuration in place? It should work....

**Comment by rachelwhitton** [comment]
@lauren-kelly So I made some progress.. 
screen_shot_2017-05-25_at_6_52_58_pm

Just doing this triggered New Relic monitor alerts with the following error: `HTTPError: Server sent us too many redirects (20)`

The response looked like: 
```
$ curl -I https://www.rachelwhitton.com/
HTTP/1.1 301 Moved Permanently
Date: Thu, 25 May 2017 23:51:50 GMT
Connection: keep-alive
Cache-Control: max-age=3600
Expires: Fri, 26 May 2017 00:51:50 GMT
Location: http://www.rachelwhitton.com/
Server: cloudflare-nginx
CF-RAY: 364c67ab5c135903-DFW
```

After enabling Strict SSL in Cloudflare, New Relic monitor recovered and the response looked like: 
```
$ curl -I https://www.rachelwhitton.com/
HTTP/1.1 200 OK
Date: Thu, 25 May 2017 23:52:02 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Set-Cookie: __cfduid=db45f04a23eceb6c8f317a5b8e5a0a9231495756322; expires=Fri, 25-May-18 23:52:02 GMT; path=/; domain=.rachelwhitton.com; HttpOnly
Cache-Control: public, max-age=600
Link: <https://www.rachelwhitton.com/wp-json/>; rel="https://api.w.org/"
Link: <https://www.rachelwhitton.com/>; rel=shortlink
Strict-Transport-Security: max-age=15984000; includeSubDomains; preload
Surrogate-Key-Raw:
X-Pantheon-Styx-Hostname: styx636a566b
X-Pingback: https://www.rachelwhitton.com/xmlrpc.php
X-Styx-Req-Id: styx-db6856ace866f5c334f4e893607b08f9
Via: 1.1 varnish
Fastly-Debug-Digest: eb5cdf009f62423c516e2324e373ae9e320364e110526c09f63875191870088f
Via: 1.1 varnish
Age: 208
X-Served-By: cache-ord1735-ORD, cache-dfw1826-DFW
X-Cache: HIT, HIT
X-Cache-Hits: 1, 1
X-Timer: S1495756322.308558,VS0,VE1
Vary: Accept-Encoding, Cookie, Cookie
Server: cloudflare-nginx
CF-RAY: 364c67f5692b1171-DFW
```

Pantheon Dashboard shows action required 

cc @ari-gold

**Comment by lauren-kelly** [comment]
:+1:  Awesome @rachelwhitton - Thank you!

**Pull request merged**